### ISSUE TITLE
Key error when scanning repository

### ISSUE THREAD

**Issue by pb82** [ansible/ansible-lint#3973]
##### Summary

When running `ansible-lint --fix` against https://github.com/ansible-collections/amazon.aws.git, a `KeyError` is raised and ansible-lint fails:

```
Traceback (most recent call last):
  File "/home/peter/.local/bin/ansible-lint", line 8, in <module>
    sys.exit(_run_cli_entrypoint())
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/peter/.local/lib/python3.12/site-packages/ansiblelint/__main__.py", line 402, in _run_cli_entrypoint
    sys.exit(main(sys.argv))
             ^^^^^^^^^^^^^^
  File "/home/peter/.local/lib/python3.12/site-packages/ansiblelint/__main__.py", line 379, in main
    fix(runtime_options=options, result=result, rules=rules)
  File "/home/peter/.local/lib/python3.12/site-packages/ansiblelint/__main__.py", line 248, in fix
    _do_transform(result, options)
  File "/home/peter/.local/lib/python3.12/site-packages/ansiblelint/__main__.py", line 203, in _do_transform
    transformer.run()
  File "/home/peter/.local/lib/python3.12/site-packages/ansiblelint/transformer.py", line 118, in run
    self._do_transforms(file, ruamel_data or data, file_is_yaml, matches)
  File "/home/peter/.local/lib/python3.12/site-packages/ansiblelint/transformer.py", line 156, in _do_transforms
    match.rule.transform(match, file, data)
  File "/home/peter/.local/lib/python3.12/site-packages/ansiblelint/rules/name.py", line 179, in transform
    ] = f"{target_task['name'][:1].upper()}{target_task['name'][1:]}"
           ~~~~~~~~~~~^^^^^^^^
  File "/home/peter/.local/lib/python3.12/site-packages/ruamel/yaml/comments.py", line 851, in __getitem__
    return ordereddict.__getitem__(self, key)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'name'
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 6.22.1 using ansible-core:2.16.2 ansible-compat:4.1.10 ruamel-yaml:0.18.5 ruamel-yaml-clib:0.2.7
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

1. Clone https://github.com/ansible-collections/amazon.aws.git
2. Run `ansible-lint --fix` in that repository directory

##### Desired Behavior

ansible-lint should not fail with a KeyError and exit code 1.

##### Actual Behavior

ansible-lint fails with a KeyError, exit code is 1.

```paste below
[peter@fedora amazon.aws]$ ansible-playbook --syntax-check playbook
ERROR! the playbook: playbook could not be found
```

[minimum complete verifiable example]: see above

**Comment by ganeshrn** [comment]
The issue is seeing while scanning the repository using Ansible Code Bot.

**Comment by TamiTakamiya** [comment]
The error (it may not be the only one, though) occurs with https://github.com/ansible-collections/amazon.aws/blob/main/tests/integration/targets/elb_classic_lb/tasks/schema_change.yml
